src/Plugins/RpcServer/RpcServer.cs

```diff
@@ -288,25 +352,120 @@ private async Task<JObject> ProcessRequestAsync(HttpContext context, JObject req
             {
                 return CreateErrorResponse(request["id"], RpcError.InvalidParams.WithData(ex.Message));
             }
-            catch (Exception ex)
+            catch (Exception ex) when (ex is not RpcException)
             {
 #if DEBUG
                 return CreateErrorResponse(request["id"], RpcErrorFactory.NewCustomError(ex.HResult, ex.Message, ex.StackTrace));
 #else
-                return CreateErrorResponse(request["id"], RpcErrorFactory.NewCustomError(ex.HResult, ex.Message));
+        return CreateErrorResponse(request["id"], RpcErrorFactory.NewCustomError(ex.HResult, ex.Message));
 #endif
             }
         }
 
         public void RegisterMethods(object handler)
         {
-            foreach (MethodInfo method in handler.GetType().GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
+            foreach (var method in handler.GetType().GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance))
             {
-                RpcMethodAttribute attribute = method.GetCustomAttribute<RpcMethodAttribute>();
-                if (attribute is null) continue;
-                string name = string.IsNullOrEmpty(attribute.Name) ? method.Name.ToLowerInvariant() : attribute.Name;
-                methods[name] = method.CreateDelegate<Func<JArray, object>>(handler);
+                var attribute = method.GetCustomAttribute<RpcMethodAttribute>();
+                var attributeWithParams = method.GetCustomAttribute<RpcMethodWithParamsAttribute>();
+                if (attribute is null && attributeWithParams is null) continue;
```

Comment:
This is because i can not do all the convert in one pr, that will make this pr too large to review. thus still keep old attribute. Will be removed in the last pr of this task.